### PR TITLE
MGP added

### PR DESCRIPTION
**Pull request by repriem**
Add gradient based optimization of the hyperparameters
Add MGP
Add several kernels (Matern 3/2 and 5/2)

### CODE REVIEW

**Comment by repriem** [comment]
Doc is not possible to merge because of the sphinx_auto_embed doc generation

File: setup.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 This package is distributed under New BSD license.
 """
 
+from __future__ import print_function
```
Comment (relf) [inline setup.py:9]: to be removed

File: smt/applications/tests/test_mfk.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 @author: m.meliani
 """
 
+from __future__ import print_function, division
```
Comment (relf) [inline smt/applications/tests/test_mfk.py:8]: to be removed

File: smt/surrogate_models/tests/test_surrogate_model_examples.py [hunk 3]
```diff
@@ -8,7 +8,8 @@
 
 import matplotlib
 
-matplotlib.use("Agg")
+# matplotlib.use("Agg")
```
Comment (relf) [inline smt/surrogate_models/tests/test_surrogate_model_examples.py:11]: I think  matplotlib.use("Agg") should be enabled

**Review by relf — APPROVED** [review]
Ok (minor cosmetic changes)

File: smt/surrogate_models/krg_based.py [hunk 4]
```diff
@@ -895,6 +895,8 @@ def grad_minus_reduced_likelihood_function(log10t):
                             stop += 1
                             if incr != 0:
                                 return
+                            if stop > max_retry:
+                                raise ValueError('%d attempt to train the model failed' %max_retry)
```
Comment (relf) [inline smt/surrogate_models/krg_based.py:899]: attempts and would you address previous review minor requested change?

**Review by relf — CHANGES_REQUESTED** [review]

File: smt/surrogate_models/krg_based.py (hunk 4 above)
Comment (repriem) [reply smt/surrogate_models/krg_based.py:899]: Ok no problem

**Review by relf — APPROVED** [review]

**Pull request merged**